### PR TITLE
fix(Storefront): STRF-5072 Cornerstone not showing Sale badge 

### PR DESCRIPTION
**Pull request by yurytut1993**
#### What?

@bc-themes 
Fixed Cornerstone not showing Sale badge in list view. 
In your page builder you have theme styles. In Global section you need to scroll opened modal window to Products paragraph. Here you can select display style list or grid. When grid is selected Sale Badges are seen on products that have a discount. But if you select list, product items will not have Sale Badges.
For now it is fixed - the user can see Sale Badges on items this discount in list view and in a grid view.
I added code that checks is this item on sale and need badge in html file of product list item. Also I updated styles for badges displaing in product list item
 
#### Tickets / Documentation

Add links to any relevant tickets and documentation.
https://jira.bigcommerce.com/browse/STRF-5072

#### Screenshots (if appropriate)

Attach images or add image links here.

Screenshot 2020-06-16 at 14 59 28
Screenshot 2020-06-16 at 15 00 48

### CODE REVIEW

**Comment by junedkazi** [comment]
@yurytut1993 can you please add a better title to reflect what is being done as part of this PR.

**Comment by yurytut1993** [comment]
> @yurytut1993 can you please add a better title to reflect what is being done as part of this PR.

its done. Please check and ask me if you need more information

File: templates/components/products/list-item.html [hunk 1]
```diff
@@ -4,6 +4,22 @@
     <article class="listItem">
 {{/if}}
     <figure class="listItem-figure">
+        {{#or price.non_sale_price_with_tax price.non_sale_price_without_tax}}
+        {{#if theme_settings.product_sale_badges '===' 'sash'}}
+            <div class="sale-flag-sash">
+                <span class="sale-text">On Sale!</span>
```
Comment (junedkazi) [inline templates/components/products/list-item.html:10]: Is there a reason we need these span tags ? Also I this is going to be surfaced in the theme the I think we need to move the text into `schema.json` and `config.json` so we merchants can modify them.

thoughts @bookernath @bc-as ??

Comment (yurytut1993) [reply templates/components/products/list-item.html:10]: span tags used for text styles and aligning. We have already used this HTML construction for sale badges in products grid. Of course we can add possibility to our merchants to edit sale badge text. But I suppose that we need to add an input for it control panel firstly. To allow merchants set up it. In this pull request I just wanted to fix the bug with lack of sale badge in products list

Comment (bc-as) [reply templates/components/products/list-item.html:10]: @junedkazi we historically haven't exposed the text on those badges to be configurable, but it is a good idea.  I'll open another ticket to address that feature.

Comment (bc-as) [reply templates/components/products/list-item.html:10]: https://jira.bigcommerce.com/browse/BCTHEME-61 for sales badge text

**Pull request merged**